src/Evolution/Systems/Ccz4/Tags.hpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <cstddef>
+#include <string>
+
+#include "DataStructures/DataBox/Tag.hpp"
+#include "DataStructures/Tensor/TypeAliases.hpp"
+#include "Evolution/Systems/Ccz4/TagsDeclarations.hpp"
+#include "Evolution/Tags.hpp"
+#include "Options/Options.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags.hpp"
+#include "PointwiseFunctions/GeneralRelativity/Tags/Conformal.hpp"
+
+namespace Ccz4 {
+namespace Tags {
+/*!
+ * \brief The conformal factor that rescales the spatial metric
+ *
+ * \details If \f$\gamma_{ij}\f$ is the spatial metric, then we define
+ * \f$\phi = (det(\gamma_{ij}))^{-1/6}\f$.
+ */
+template <typename DataType>
+struct ConformalFactor : db::SimpleTag {
+  using type = Scalar<DataType>;
+};
+
+/*!
+ * \brief The conformally scaled spatial metric
+ *
+ * \details If \f$\phi\f$ is the conformal factor and \f$\gamma_{ij}\f$ is the
+ * spatial metric, then we define
+ * \f$\bar{\gamma}_{ij} = \phi^2 \gamma_{ij}\f$.
+ */
+template <size_t Dim, typename Frame, typename DataType>
+using ConformalMetric =
+    gr::Tags::Conformal<gr::Tags::SpatialMetric<Dim, Frame, DataType>, 2>;
+
+/*!
+ * \brief The conformally scaled inverse spatial metric
+ *
+ * \details If \f$\phi\f$ is the conformal factor and \f$\gamma^{ij}\f$ is the
+ * inverse spatial metric, then we define
+ * \f$\bar{\gamma}^{ij} = \phi^2 \gamma^{ij}\f$.
+ */
+template <size_t Dim, typename Frame, typename DataType>
+using InverseConformalMetric =
+    gr::Tags::Conformal<gr::Tags::InverseSpatialMetric<Dim, Frame, DataType>,
+                        2>;
+
+/*!
+ * \brief The determinant of the conformally scaled spatial metric
+ */
+template <typename DataType>
+struct DetConformalMetric : db::SimpleTag {
```

Comment:
Forgive me, because I'm not very numerical relativity literate, so this may be misguided, but I thought since eq. 12a evolves the conformal spatial metric and because the determinant of it is in the last term of that eq., I thought maybe the determinant might not always be 1. Nils Deppe was explaining to me that this last term in 12a is a constraint, so I thought that, perhaps, the determinant isn't always actually equal to 1, numerically. Let me know what you think! I don't really have any NR domain knowledge

Reply:
If it should always be 1 numerically, then I agree that there isn't a need for a tag

Reply:
So this ^ was just my reasoning for making the tag in the first place, but if you think the tag is unnecessary, I'll remove it :+1:

Reply:
Yea that sounds right. The determinant will not be identically 1 numerically, and the term in Eq. (12a) is responsible for driving the determinant towards 1. Since you only need tags for quantities that you store in the DataBox, or to identify quantities e.g. to compute them analytically, you will probably not need one for the determinant. My guess is that you will compute the determinant on-the-fly when evaluating the r.h.s of Eq. (12a). So I suggest you remove it for now, and if you ever need it it's easy enough to add back in.

Reply:
Sounds good, I'll remove it